cli/azd/cmd/auth_token.go

```diff
@@ -76,8 +86,36 @@ func (a *authTokenAction) Run(ctx context.Context) (*actions.ActionResult, error
 
 	var cred azcore.TokenCredential
 
+	// 1) flag --tenant-id is the highest priority. If it is not use, azd will check if subscriptionId is set as env var
+	tenantId := a.flags.tenantID
+	// 2) try to resolve tenant id when AZURE_SUBSCRIPTION_ID is set as system env
+	if tenantId == "" {
+		if subIdAtSystemEnv, found := os.LookupEnv(environment.SubscriptionIdEnvVarName); found {
+			if resolvedTenantId, err := a.subResolver.LookupTenant(ctx, subIdAtSystemEnv); err == nil {
+				tenantId = resolvedTenantId
+			} else {
+				log.Println("Found AZURE_SUBSCRIPTION_ID in system env, but azd couldn't find the Azure directory where" +
+					" it belongs. The AZURE_SUBSCRIPTION_ID is ignored.")
```

Comment:
I think this would be an error?

1. A user has explicitly set `AZURE_SUBSCRIPTION_ID`
2. `azd` could not find resolved from the tenant ID

I can't see why user would have expected `azd` to ignore and went on to find it differently.

Reply:
I thought about it. 
I was thinking it could be easier for a user to set an invalid value to the env-var to disable it, instead of removing the env-var completely.  But it might be a bias on my side, as that's what I'd do. 

I'd make azd to fail, ok.